SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularityMesosTaskBuilder.java

```diff
@@ -239,6 +239,14 @@ private void prepareContainerInfo(final Offer offer, final SingularityTaskId tas
             dockerInfoBuilder.addPortMappings(maybePortMapping.get());
           }
         }
+      } else if (configuration.getNetworkConfiguration().isDefaultPortMapping() && dockerInfo.get().getPortMappings().isEmpty() && ports.isPresent()) {
```

Comment:
Might also need to check that network mode is `BRIDGE` here, if it's `HOST` or `NONE` and you specify mappings believe that will end up at least in a `TASK_FAILED` if not a `TASK_ERROR`. Other than that looks good